### PR TITLE
joystickwake: init at 0.2.4

### PR DESCRIPTION
**Pull request by bertof**
###### Motivation for this change
Add package for [joystickwake](https://github.com/foresto/joystickwake), an utility for disabling screensavers and auto-lock while any input is received from a gamepad. This solves the annoying issue of auto-lock locking the session while playing games.

###### Things done
- Add package definition in `pkgs/tools/games/joystickwake/default.nix`
- Add package to the `all-packages.nix` file
- Add myself as maintainer

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [x] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [21.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md#generating-2111-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by r-rmcgibbo** [comment]
Result of `nixpkgs-review pr 127879` at 949844cf run on aarch64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built successfully:</summary>
  <ul>
    <li>joystickwake</li>
  </ul>
</details>
<details>
  <summary>1 suggestion:</summary>
  <ul>
    <li>warning: <a href="https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/no-python-tests.md">no-python-tests</a>

Test runner could not discover any test cases: ‘Ran 0 tests in 0.000s’
Near pkgs/tools/games/joystickwake/default.nix:20:0:
```
   |
20 |     description = "A joystick-aware screen waker";
   | ^
```
</li>
  </ul>
</details>

File: maintainers/maintainer-list.nix [hunk 1]
```diff
@@ -1220,6 +1220,16 @@
     github = "benneti";
     githubId = 11725645;
   };
+  bertof = {
```
Comment (SuperSandro2000) [inline maintainers/maintainer-list.nix:1223]: Please put this into a separate commit with the message ``maintainers: add NAME`` and sort it alphabetically.

File: pkgs/tools/games/joystickwake/default.nix [hunk 2]
```diff
@@ -0,0 +1,31 @@
+{ lib, python3, fetchFromGitHub, withX11 ? true }:
+python3.pkgs.buildPythonApplication rec {
```
Comment (SuperSandro2000) [inline pkgs/tools/games/joystickwake/default.nix:2]:
```suggestion
{ lib, python3, fetchFromGitHub }:

python3.pkgs.buildPythonApplication rec {
```

File: pkgs/tools/games/joystickwake/default.nix [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+  src = fetchFromGitHub {
+    owner = "foresto";
+    repo = pname;
+    rev = "v${version}";
+    sha256 = "0j8xwfmzzmc9s88zvzc3lv67821r6x28vy6vli3srvx859wprppd";
+  };
+
+  buildInputs = with python3.pkgs;
+    ([ pyudev ] ++ (lib.optional withX11 xlib));
+
```
Comment (SuperSandro2000) [inline pkgs/tools/games/joystickwake/default.nix:15]:
```suggestion
```

File: pkgs/tools/games/joystickwake/default.nix [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+  propagatedBuildInputs = with python3.pkgs;
+    ([ pyudev ] ++ (lib.optional withX11 xlib));
```
Comment (SuperSandro2000) [inline pkgs/tools/games/joystickwake/default.nix:17]:
```suggestion
  propagatedBuildInputs = with python3.pkgs; [ pyudev xlib ];
```

File: pkgs/tools/games/joystickwake/default.nix [hunk 5]
```diff
@@ -0,0 +1,31 @@
...
+      Linux gamers often find themselves unexpectedly staring at a blank screen, because their display server fails to recognize game controllers as input devices, allowing the screen blanker to activate during gameplay.
+      This program works around the problem by temporarily disabling screen blankers when joystick activity is detected.
+    '';
+    homepage = "https://github.com/foresto/joystickwake";
+    maintainers = with maintainers; [ bertof ];
+    license = licenses.mit;
+    platforms = platforms.linux;
+  };
+
+}
```
Comment (SuperSandro2000) [inline pkgs/tools/games/joystickwake/default.nix:31]:
```suggestion
  };
}
```

**Review by SuperSandro2000 — CHANGES_REQUESTED** [review]

File: maintainers/maintainer-list.nix (hunk 1 above)
Comment (bertof) [reply maintainers/maintainer-list.nix:1223]: Ok

**Comment by bertof** [comment]
I applied all the proposed changes

**Comment by bertof** [comment]
Just a question. From the application readme `xlib` is optional and this may also work on Wayland. Is there any way of indicating it?

**Comment by SuperSandro2000** [comment]
> Just a question. From the application readme `xlib` is optional and this may also work on Wayland. Is there any way of indicating it?

No, not really.
